src/cpp/include/nodegui/QtWidgets/QTreeWidget/ntreewidget.hpp

```diff
@@ -20,5 +21,35 @@ class NTreeWidget : public QTreeWidget, public NodeWidget {
       Napi::HandleScope scope(env);
       this->emitOnNode.Call({Napi::String::New(env, "itemSelectionChanged")});
     });
+
+    QObject::connect(
+        this, &QTreeWidget::itemClicked,
+        [=](QTreeWidgetItem* item, int column) {
+          Napi::Env env = this->emitOnNode.Env();
+          Napi::HandleScope scope(env);
+
+          // Disable deletion of the native instance for these by passing true
+          Napi::Object itemWrap = QTreeWidgetItemWrap::constructor.New(
+              {Napi::External<QTreeWidgetItem>::New(env, item),
+               Napi::Boolean::New(env, true)});
```

Comment:
In the case of QTreeWidgetItem, since it doesnt inherit from things like QObject and has a valid copy constructor https://doc.qt.io/archives/qt-4.8/qtreewidgetitem.html#QTreeWidgetItem-9

We can safely allow deletion. 
So here 
instead of 
```
Napi::Object itemWrap = QTreeWidgetItemWrap::constructor.New(
              {Napi::External<QTreeWidgetItem>::New(env, item),
               Napi::Boolean::New(env, true)});
```

you can do 
```

Napi::Object itemWrap = QTreeWidgetItemWrap::constructor.New(
              {Napi::External<QTreeWidgetItem>::New(env, new QTreeWidgetItem(item))});
```
This would be much safer memory wise since it would get auto deleted when the corresponding js object gets gc'd.

Reply:
Thanks for the hint! My c++ is a bit rusty, especially in regard to pointers and memory. I will implement the suggested changes.

Reply:
@master-atul I've been changing the code as discussed but ran into an annoying issue. Maybe you know a solution to this problem. When using the copy constructor new items magically appear in the QTreeWidget, without any code present for adding new items at runtime. 
[Please, take a look at this video of the problem.](https://www.dropbox.com/s/ew66gdaq4o37yhj/error-with-copy.mp4?dl=0)

When reverting the code back to its original state, e.g. **not** using the copy constructor, the problem does not occur and everything is working as expected. 
[You can find a video of a working version here.](https://www.dropbox.com/s/qfnihxv138gln3q/no-error-with-pointer.mp4?dl=0)

Something's wrong, but I'm not able to figure out what exactly. Are any additional changes required if we would stick to the current implementation? Regarding memory for example?

Reply:
Nevermind, found the cause. Forgot to dereference the pointer which caused the underlying children to be copied as well. Changing to
```c++
Napi::Object itemWrap = QTreeWidgetItemWrap::constructor.New(
              {Napi::External<QTreeWidgetItem>::New(
                  env, new QTreeWidgetItem(*item))});
```
solves the issue.